include/envoy/stream_info/stream_info.h

```diff
@@ -64,6 +65,50 @@ enum ResponseFlag {
   LastFlag = StreamIdleTimeout
 };
 
+struct ResponseCodeDetailValues {
+  const std::string SET_BY_UPSTREAM = "set_by_upstream";
+  // Envoy is doing non-streaming proxying, and the request payload exceeded
+  // configured limits.
+  const std::string REQUEST_PAYLOAD_TOO_LARGE = "request_payload_too_large";
+  // Envoy is doing non-streaming proxying, and the response payload exceeded
+  // configured limits.
+  const std::string RESPONSE_PAYLOAD_TOO_LARGE = "response_payload_too_large";
+  // The per-stream keepalive timeout was exceeded.
+  const std::string STREAM_IDLE_TIMEOUT = "stream_idle_timeout";
+  // The per-stream total request timeout was exceeded
+  const std::string REQUEST_OVERALL_TIMEOUT = "request_overall_timeout";
+  // The request was rejected due to the Overload Manager reaching configured resource limits.
+  const std::string OVERLOAD = "overload";
+  // The HTTP/1.0 or HTTP/0.9 request was rejected due to HTTP/1.0 support not being configured.
+  const std::string LOW_VERSION = "low_version";
+  // The request was rejected due to the Host: or :authority field missing
+  const std::string MISSING_HOST = "missing_host_header";
+  // The request was rejected due to the Host: or :authority field missing
+  const std::string REQUEST_HEADERS_TOO_LARGE = "request_headers_too_large";
+  // The request was rejected because path normalization was configured on and failed, probably due
+  // to an invalid path.
+  const std::string PATH_NORMALIZATION_FAILED = "path_normalization_failed";
+  // The request was rejected because it attempted an unsupported upgrade.
+
+  // TODO(reviewers) do we want a standard of filter etc. prefix or unique IDs enough?
```

Comment:
Could we simply use filter's well know name? Unique ID might make debuggability harder.